### PR TITLE
Refactor benchmarks

### PR DESCRIPTION
**Pull request by officialasishkumar**
### :pencil: Description

**Type:** :beetle: `bugfix` 

- fix the benchmark action of PR
- changed the name of the classes and files
- removed the benchmarks which are not important. 

### :pushpin: Resources

Examples, notebooks, and links to useful references.

### :vertical_traffic_light: Testing

How did you test these changes?

- [ ] Testing pipeline
- [ ] Other method (describe)
- [ ] My changes can't be tested (explain why)

### :ballot_box_with_check: Checklist

- [ ] I requested two reviewers for this pull request
- [ ] I updated the documentation according to my changes
- [ ] I built the documentation by applying the `build_docs` label

> **Note:** If you are not allowed to perform any of these actions, ping (@) a contributor.

### CODE REVIEW

File: benchmarks/transport_montecarlo_formal_integral_p.py
Comment (andrewfullard) [inline benchmarks/transport_montecarlo_formal_integral_p.py:1]: Why is this called `formal_integral_p`?

File: benchmarks/transport_montecarlo_formal_integral_p.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    @parameterize(
+        {
+            "Parameters": [
+                {
+                    "nu": 1e14,
+                    "temperature": 1e4,
+                },
+                {
+                    "nu": 0,
+                    "temperature": 1,
```
Comment (andrewfullard) [inline benchmarks/transport_montecarlo_formal_integral_p.py:28]: These aren't physical, I don't think they are worth parametrizing

File: benchmarks/transport_montecarlo_formal_integral_p.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+                },
+                {
+                    "nu": 1,
+                    "temperature": 1,
```
Comment (andrewfullard) [inline benchmarks/transport_montecarlo_formal_integral_p.py:32]: Same here

File: benchmarks/transport_montecarlo_formal_integral_p.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    """
+    Class to benchmark the numba formal integral function.
+    """
+
+    @parameterize(
+        {
+            "Parameters": [
+                {
+                    "nu": 1e14,
+                    "temperature": 1e4,
```
Comment (andrewfullard) [inline benchmarks/transport_montecarlo_formal_integral_p.py:24]: This is okay

File: benchmarks/transport_montecarlo_formal_integral_p.py
Comment (officialasishkumar) [reply benchmarks/transport_montecarlo_formal_integral_p.py:1]: The `_p` was initially for the parallelism thing. I have removed it now.

**Comment by officialasishkumar** [comment]
Action run: https://github.com/officialasishkumar/tardis/actions/runs/10041925413/job/27751118212?pr=53

**Comment by officialasishkumar** [comment]
The benchmark run is successful in these changes. 

link: https://paste.opensuse.org/pastes/0e80fd45e5c1

**Comment by andrewfullard** [comment]
I notice the before and after commits are still the same in the bot comment. Should that have been resolved here?

**Comment by officialasishkumar** [comment]
yes

**Comment by officialasishkumar** [comment]
new action run on the existing benchmarks.yml -> https://github.com/officialasishkumar/tardis/actions/runs/10077232306/job/27859281972

**Comment by officialasishkumar** [comment]
I have ironed out all the bugs. So this can be merged to get the benchmarks working. 

cc: @andrewfullard

File: tardis/transport/montecarlo/single_packet_loop.py [hunk 4]
```diff
@@ -61,6 +62,9 @@ def single_packet_loop(
         This function does not return anything but changes the r_packet object
         and if virtual packets are requested - also updates the vpacket_collection
     """
+
+    for i in range(200):
+        print("hello")
```
Comment (andrewfullard) [inline tardis/transport/montecarlo/single_packet_loop.py:67]: Please remove this

File: tardis/transport/montecarlo/single_packet_loop.py [hunk 5]
```diff
@@ -30,6 +30,7 @@
 )
 from tardis.transport.montecarlo.vpacket import trace_vpacket_volley
 
+import time
```
Comment (andrewfullard) [inline tardis/transport/montecarlo/single_packet_loop.py:33]: Please remove this

**Review by andrewfullard — CHANGES_REQUESTED** [review]
Good job fixing the benchmarks, please remove the test items

File: tardis/transport/montecarlo/single_packet_loop.py (hunk 4 above)
Comment (officialasishkumar) [reply tardis/transport/montecarlo/single_packet_loop.py:67]: sorry, my bad. I will fix it.

File: tardis/transport/montecarlo/single_packet_loop.py [hunk 6]
```diff
@@ -61,6 +61,7 @@ def single_packet_loop(
         This function does not return anything but changes the r_packet object
         and if virtual packets are requested - also updates the vpacket_collection
     """
+
```
Comment (andrewfullard) [inline tardis/transport/montecarlo/single_packet_loop.py:64]: Excess line change

File: tardis/transport/montecarlo/single_packet_loop.py (hunk 5 above)
Comment (officialasishkumar) [reply tardis/transport/montecarlo/single_packet_loop.py]: removed

File: tardis/transport/montecarlo/single_packet_loop.py (hunk 4 above)
Comment (officialasishkumar) [reply tardis/transport/montecarlo/single_packet_loop.py]: removed

File: tardis/transport/montecarlo/single_packet_loop.py (hunk 6 above)
Comment (officialasishkumar) [reply tardis/transport/montecarlo/single_packet_loop.py]: removed

**Review by andrewfullard — APPROVED** [review]

**Pull request merged**